src/cli/expand-patterns.js

```diff
@@ -84,10 +84,9 @@ function* expandPatternsInternal(context) {
       } else if (stat.isDirectory()) {
         entries.push({
           type: "dir",
-          glob:
-            escapePathForGlob(fixWindowsSlashes(pattern)) +
-            "/" +
-            getSupportedFilesGlob(),
+          glob: `${escapePathForGlob(
+            fixWindowsSlashes(pattern)
+          )}/${getSupportedFilesGlob()}`,
```

Comment:
But we really should improve prettier here...

Reply:
How exactly?

Reply:
More smart logic for breaks, we have a circular problem - prettier format templates ugly in some cases, therefore developers complain that the `eslint` rule do code ugly and disable it. I agree it can be less aggressive, but prettier is not good too. And I agree that it is not so easy, but I am sure that we can find a solution.

Reply:
We can't just say `eslint` rule is bad don't use it, in theory even if we discuss with eslint developers and do it less aggressive, there are still a lot of ugly cases, we can look at them in our issues and some snapshots

Reply:
Some code just can't look pretty no matter how you try to format it. Luckily the language has alternative constructions that can help in such cases. E.g. `+` for concatenation.

Reply:
But in most cases it can look better, we should get more benefits than ugly, that's why we are working on `prettier`

Reply:
What you're saying is very abstract. Let's be more specific and discuss expression with this structure: `<long expression 1> + "/" + <long expression 2>`. I say that no matter how you (or Prettier) try to format such expressions, in general case concatenation would look better than a template literal.

This specific case you commented
```js
glob: `${escapePathForGlob(
  fixWindowsSlashes(pattern)
)}/${getSupportedFilesGlob()}`
```
can be improved by not wrapping the expression at all:
```js
glob: `${escapePathForGlob(fixWindowsSlashes(pattern))}/${getSupportedFilesGlob()}`
```
with spaces it can become a bit better probably (looks unusual though):
```js
glob: `${ escapePathForGlob(fixWindowsSlashes(pattern)) }/${ getSupportedFilesGlob() }`
```
But this strategy won't work at all if the expressions are longer. So we need to break lines. But how exactly? The current version of line breaking doesn't look good to you, right? How else can it be done?
```js
// 1
glob: `${escapePathForGlob(fixWindowsSlashes(pattern))}/${
  getSupportedFilesGlob()}`

// 2
glob: `${
  escapePathForGlob(fixWindowsSlashes(pattern))
}/${
  getSupportedFilesGlob()
}`
```

Do these look better?

Reply:
Honestly I prefer do not break here, i.e.
```js
glob: `${escapePathForGlob(fixWindowsSlashes(pattern))}/${getSupportedFilesGlob()}`
```

But I agree this example show how different can it be formatted, I don't have strong opinion here, but moving `fixWindowsSlashes(pattern)` argument for `escapePathForGlob` call the the next line looks quite ugly

Reply:
> Honestly I prefer do not break here

Me too, but as I wrote, it's not a generic solution. What if the expressions are longer? Are we going to have 1000 characters long lines? Prettier needs generic solutions that can used algorithmically.